charts/connectors/templates/deployment.yaml

```diff
@@ -0,0 +1,204 @@
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: {{ .Values.deployment.podAntiAffinity.weight | int64 }}
+              podAffinityTerm:
+                topologyKey: {{ .Values.deployment.podAntiAffinity.topologyKey }}
+                labelSelector:
+                  matchLabels: {{ include "connectors-pod-labels" . | nindent 20 }}
+      {{- else if eq .Values.deployment.podAntiAffinity.type "custom" }}
+          {{- toYaml .Values.deployment.podAntiAffinity.custom | nindent 10 }}
+      {{- end }}
+  {{- end }}
+      serviceAccountName: {{ template "connectors.fullname" . }}
+      containers:
+        - name: connectors-cluster
+          image: {{ .Values.image.repository }}:{{ template "connectors.tag" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          securityContext:
+  {{- with .Values.container.securityContext }}
+    {{- toYaml . | nindent 12 }}
+  {{- end }}
+          env:
+            - name: CONNECT_CONFIGURATION
+              value: |
+                key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+                value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+                group.id={{ .Values.connectors.groupID }}
+                offset.storage.topic={{ .Values.connectors.storage.topic.offset }}
+                config.storage.topic={{ .Values.connectors.storage.topic.config }}
+                status.storage.topic={{ .Values.connectors.storage.topic.status }}
+                offset.storage.redpanda.remote.read={{ .Values.connectors.storage.remote.read.offset }}
+                offset.storage.redpanda.remote.write={{ .Values.connectors.storage.remote.write.offset }}
+                config.storage.redpanda.remote.read={{ .Values.connectors.storage.remote.read.config }}
+                config.storage.redpanda.remote.write={{ .Values.connectors.storage.remote.write.config }}
+                status.storage.redpanda.remote.read={{ .Values.connectors.storage.remote.read.status }}
+                status.storage.redpanda.remote.write={{ .Values.connectors.storage.remote.write.status }}
+                offset.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.offset }}
+                config.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.config }}
+                status.storage.replication.factor={{ .Values.connectors.storage.replicationFactor.status }}
+                producer.linger.ms={{ .Values.connectors.producerLingerMS }}
+                producer.batch.size={{ .Values.connectors.producerBatchSize }}
+                config.providers=file,secretsManager
+                config.providers.file.class=org.apache.kafka.common.config.provider.FileConfigProvider
+                {{- if .Values.connectors.secretManager.enabled }}
+                config.providers.secretsManager.class=com.github.jcustenborder.kafka.config.aws.SecretsManagerConfigProvider
```

Comment:
this is for AWS only, for GCP config is different. Please also note the providers were forked and the classes changed:

- `config.providers.secretsManager.class=com.redpanda.connectors.config.provider.aws.SecretsManagerConfigProvider` for AWS

- `config.providers.secretsManager.class=com.redpanda.connectors.config.provider.gcp.SecretManagerConfigProvider` for GCP